src/lib.rs

```diff
@@ -67,19 +68,43 @@ pub struct MiriConfig {
     pub args: Vec<String>,
 
     // The seed to use when non-determinism is required (e.g. getrandom())
-    pub seed: Option<u64>
+    pub seed: Option<u64>,
 }
 
+
 // Used by priroda.
 pub fn create_ecx<'a, 'mir: 'a, 'tcx: 'mir>(
     tcx: TyCtxt<'a, 'tcx, 'tcx>,
     main_id: DefId,
     config: MiriConfig,
 ) -> EvalResult<'tcx, InterpretCx<'a, 'mir, 'tcx, Evaluator<'tcx>>> {
+
+
+    // We build up the layout of '*mut &mut dyn core::panic::BoxMeUp'
+    let box_me_up_did = helpers::resolve_did(tcx, &["core", "panic", "BoxMeUp"])?;
+    let traits = &[ExistentialPredicate::Trait(ExistentialTraitRef {
+        def_id: box_me_up_did,
+        substs: List::empty()
+    })];
+
+    let me_mut_dyn = tcx.mk_dynamic(
+        ty::Binder::dummy(tcx.intern_existential_predicates(traits)),
+        &RegionKind::ReErased
+    );
+
+    let me_mut_ref = tcx.mk_mut_ref(&RegionKind::ReErased, me_mut_dyn);
+    let me_mut_raw = tcx.mk_mut_ptr(me_mut_ref);
+    let box_me_up_layout = tcx.layout_of(ParamEnv::empty().and(me_mut_raw))
+        .map_err(|layout| InterpError::Layout(layout))?;
+
+    let cached_types = CachedTypes {
+        box_me_up_layout
+    };
```

Comment:
I still doubt that this is worth caching. We don't panic all the time and we don't need to be extremely fast doing so.

Either way though please move building that type into a helper function.